cloud/cluster/cluster.go

```diff
@@ -1,51 +1,121 @@
 package cluster
 
 import (
-	"fmt"
-	"io"
+	"sort"
 )
 
-// NodeUpdate represents a change to the cluster
-type NodeUpdate struct {
-	UpdateType NodeUpdateType
-	Id         NodeId
-	Node       Node // Only set for adds
+// Cluster represents a cluster of Nodes.
+
+type Cluster struct {
+	State    *State
+	reqCh    chan interface{}
+	updateCh chan []NodeUpdate
+	StateCh  chan []Node
+	subs     []chan []NodeUpdate
 }
 
-func (u *NodeUpdate) String() string {
-	return fmt.Sprintf("%v %v %v", u.UpdateType, u.Id, u.Node)
+func NewCluster(state []Node, updateCh chan []NodeUpdate, stateCh chan []Node) *Cluster {
```

Comment:
i was thinking that it would be nice to split up the responsibilities of the channels to be focused on either nodes or node updates, but it does simplify it a bit to only have one channel for both.